airbyte-webapp/src/packages/cloud/components/experiments/FreeConnectorProgram/hooks/useFreeConnectorProgram.ts

```diff
@@ -30,16 +32,32 @@ export const useFreeConnectorProgram = () => {
   const [userDidEnroll, setUserDidEnroll] = useState(false);
   const { formatMessage } = useIntl();
   const { registerNotification } = useNotificationService();
+  const { trackError } = useAppMonitoringService();
 
   useEffectOnce(() => {
     if (searchParams.has(STRIPE_SUCCESS_QUERY)) {
       // Remove the stripe parameter from the URL
-      setSearchParams({}, { replace: true });
-      setUserDidEnroll(true);
-      registerNotification({
-        id: "fcp/enrolled",
-        text: formatMessage({ id: "freeConnectorProgram.enroll.success" }),
-        type: ToastType.SUCCESS,
+      pollUntil(
```

Comment:
We have pretty similar code in https://github.com/airbytehq/airbyte/blob/6a7410676941e26613592dd48c52a9f1cde33e29/airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/RemainingCredits.tsx#L73-L91 Maybe it's worth switching that also over to use the `pollUntil` utility.